src/server/game/Spells/Spell.cpp

```diff
@@ -2861,7 +2861,22 @@ void Spell::prepare(SpellCastTargets const* targets, AuraEffect const* triggered
     {
         m_castItemGUID = m_CastItem->GetGUID();
         m_castItemEntry = m_CastItem->GetEntry();
-        m_castItemLevel = int32(m_CastItem->GetItemLevel(m_CastItem->GetOwner()));
+
+        Player* owner = dynamic_cast<Player*>(targets->GetUnitTarget());
+        if (m_CastItem->GetOwner() != nullptr)
```

Comment:
Possibly indeed, I'll leave this one to veteran TC devs to make the call and apply the suggested change.